src/PIL/ImageFont.py

```diff
@@ -213,6 +214,8 @@ def load_from_bytes(f):
             )
 
         if is_path(font):
+            if isinstance(font, Path):
+                font = str(font)
```

Comment:
It's more efficient to always convert it:
```suggestion
            font = str(font)
```

string: 41ns -> 30ns
Path: 72ns -> 59ns

# Test script

```python
from pathlib import Path

font_str = "my_font.ttf"
font_path = Path("my_font.ttf")


def a(font):
    if isinstance(font, Path):
        font = str(font)

def b(font):
    font = str(font)
```

# Before

```console
❯ python --version
Python 3.12.0
❯ python -m timeit -s "import bench" "bench.a(bench.font_str)"
5000000 loops, best of 5: 40.7 nsec per loop

Pillow on  disable-gil [$?] via  v3.12.0
❯ python -m timeit -s "import bench" "bench.a(bench.font_path)"
5000000 loops, best of 5: 72.4 nsec per loop
```

# After

```console
❯ python -m timeit -s "import bench" "bench.b(bench.font_str)"
10000000 loops, best of 5: 30.3 nsec per loop

Pillow on  disable-gil [$?] via  v3.12.0 took 2s
❯ python -m timeit -s "import bench" "bench.b(bench.font_path)"
5000000 loops, best of 5: 59.2 nsec per loop
```

Reply:
`font` may also be bytes, in which case `str(font)` would be incorrect.

Reply:
Fair enough, let's keep the `isinstance`.

Shall we start gradually adding type hints? (re: https://github.com/python-pillow/Pillow/issues/2625)

Something like this:

```python
from __future__ import annotations  # [this is so we can use | instead of Union in 3.9]

...

font: bytes | str | Path | None = None
```

Reply:
Ok, I've pushed a commit.